### ISSUE TITLE
ComboBox: aria-labelledby is added to list container even when label is absent

### ISSUE THREAD

**Issue by kangka29** [microsoft/fluentui#13525]
### Environment Information

- **Package version(s)**: office-ui-fabric-react@v7.117.4
- **Browser and OS versions**: all

### Issue reproduced in codepen:
https://codepen.io/justslone-the-encoder/pen/GRWVEaa?editors=1010 (updated)

#### Actual behavior:

`aria-labelledby` attribute is added to ComboBox list container even when label is not present.

![image](https://user-images.githubusercontent.com/8017073/84129654-db889d80-aa5f-11ea-96f9-085c5dc4adab.png)

#### Expected behavior:

`aria-labelledby` attribute should be available only if label props is passed.

#### Additional code information:

The component generating the issue is present at the following path:
`packages\office-ui-fabric-react\src\components\ComboBox\ComboBox.tsx`

The method `_onRenderList` does not check for presence of `label` props while adding the `aria-labelledby` attribute.

```
    return (
      <div
        id={id + '-list'}
        className={this._classNames.optionsContainer}
        aria-labelledby={id + '-label'}
        role="listbox"
      >
        {options.map(item => (onRenderItem as any)(item, this._onRenderItem))}
      </div>
    );
```

This scenario is handled for `Autofill` component inside `render` method and same should be applied to the above method.

```
              <Autofill
                data-ktp-execute-target={keytipAttributes['data-ktp-execute-target']}
                data-is-interactable={!disabled}
                componentRef={this._autofill}
                id={id + '-input'}
                className={this._classNames.input}
                type="text"
                onFocus={this._select}
                onBlur={this._onBlur}
                onKeyDown={this._onInputKeyDown}
                onKeyUp={this._onInputKeyUp}
                onClick={this._onAutofillClick}
                onTouchStart={this._onTouchStart}
                onInputValueChange={this._onInputChange}
                aria-expanded={isOpen}
                aria-autocomplete={this._getAriaAutoCompleteValue()}
                role="combobox"
                readOnly={disabled || !allowFreeform}
                aria-labelledby={label && id + '-label'}
                aria-label={ariaLabel && !label ? ariaLabel : undefined}
                aria-describedby={
                  errorMessage !== undefined
                    ? mergeAriaAttributeValues(ariaDescribedBy, keytipAttributes['aria-describedby'], errorMessageId)
                    : mergeAriaAttributeValues(ariaDescribedBy, keytipAttributes['aria-describedby'])
                }
                aria-activedescendant={this._getAriaActiveDescendantValue()}
                aria-required={required}
                aria-disabled={disabled}
                aria-owns={isOpen ? id + '-list' : undefined}
                spellCheck={false}
                defaultVisibleValue={this._currentVisibleValue}
                suggestedDisplayValue={suggestedDisplayValue}
                updateValueInWillReceiveProps={this._onUpdateValueInAutofillWillReceiveProps}
                shouldSelectFullInputValueInComponentDidUpdate={
                  this._onShouldSelectFullInputValueInAutofillComponentDidUpdate
                }
                title={title}
                preventValueSelection={!focused}
                placeholder={placeholder}
                tabIndex={tabIndex}
                {...autofill}
              />
```

**Comment by paulgildea** [comment]
@jurokapsiar, @dzearing which is the preferred/correct pattern? The behavior that's in `AutoFill` or in `ComboBox`

Also would this be considered a breaking change if we were to change this behavior?

**Comment by kangka29** [comment]
Hi, is there any update on this bug?

**Comment by kangka29** [comment]
Please update on the status for this bug fix

**Comment by jurokapsiar** [comment]
@paulgildea aria-labelledby should only point to elements that exist in the DOM. If the label is not provided, I assume that the label slot is not rendered and thus aria-labelledby should not be used. I do not think changing this would be a breaking change.

**Comment by JustSlone** [comment]
This is an actual issue, reopening

**Comment by JustSlone** [comment]
Updated issue description with fixed codepen and additional screenshot
